### DISCUSSION TITLE
add external data to vulnerability detection

### DISCUSSION THREAD

**Discussion by MathisLandreau** [wazuh/wazuh#26527] [Q&A]
I don't know how I can send software and version from a CSV file to the manager.
I need to add a list of software with their version so that vulnerability detection can analyze it. The software comes from an external provider and is given to me in the form of a CSV file. I can't get any ageing on the provider's machines... 

in short, I want to send a list of software and versions directly to the vulnerability detection function.

If you have an idea or a solution I'm interested. I can give more details if you have questions.

**Comment by Damian-Mangold** [comment]
hi @MathisLandreau. Wazuh manager does not have this functionality inherently available. To detect a vulnerability, it is necessary to follow the standard workflow: collect package information from an agent, send it to the manager, and then allow the manager to perform the vulnerability analysis based on the agent's operating system and package information.

However, if you compile the Wazuh source code, a vulnerability scanner test tool is generated, which may allow you to perform the scan you are looking for. Let me verify the feasibility of this option and outline the steps required to carry out the test.

**Reply by MathisLandreau** [reply]
Thanks, I'm waiting to hear if it's possible or not. 
Thank you very much

**Reply by MathisLandreau** [reply]
If this doesn't work, you'll need a solution to create fake agents with the operating systems sent by the service provider (e.g. a firewall).

**Reply by Damian-Mangold** [reply]
Hi @MathisLandreau. I have been investigating, running some tests, and consulting with the development team. I can confirm that the *testtool* was not designed or developed to perform this type of testing or verification in a straightforward manner. In Wazuh version 5.x, some changes are being made to the *testtool* to simplify its use, which would make it possible to carry out the kind of testing you are looking to perform.

Since improvements are being implemented in the *testtool* for version 5.x, I would appreciate it if you could [open an issue](https://github.com/wazuh/wazuh/issues/new?assignees=&labels=&projects=&template=default.md&title=) in the Wazuh repository detailing your specific requirements. This way, the development team can evaluate the possibility of incorporating this functionality in future versions.

When you create the issue, please share it here so that I can pass it to the development team.

**Reply by MathisLandreau** [reply]
Hello, 
here is the link, thank you for your help. If you need more information or help I am available. https://github.com/wazuh/wazuh/issues/26651

**Reply by Damian-Mangold** [reply]
Thank you very much. I have added the issue to the development team's project so that it can be analyzed and evaluated. Let's continue to follow up there.

**Reply by MathisLandreau** [reply]
Hello, @Damian-Mangold 
I would like to know if the situation has changed? I haven't found a solution to the problem of external information being brought into vulnerability detection. I've tried several thechniques to get around this but without success.

So I'm open to ideas. Thanks for your help.

**Comment by sebasfalcone** [comment]
Feature discussion being tracked here:
- https://github.com/wazuh/community/issues/38521
